drivers/gpio/gpio_litex.c

```diff
@@ -191,16 +189,45 @@ static int gpio_litex_port_toggle_bits(const struct device *dev,
 	return 0;
 }
 
+static void gpio_litex_irq_handler(const struct device *dev)
+{
+	const struct gpio_litex_cfg *gpio_config = DEV_GPIO_CFG(dev);
+	struct gpio_litex_data *data = dev->data;
+
+	uint8_t int_status =
+		litex_read(GPIO_EV_PENDING(dev), gpio_config->reg_size);
+
+	/* clear events */
+	litex_write(GPIO_EV_PENDING(dev), gpio_config->reg_size,
+			int_status);
+
+	gpio_fire_callbacks(&data->cb, dev, int_status);
+}
+
+static int gpio_litex_manage_callback(const struct device *dev,
+				      struct gpio_callback *callback, bool set)
+{
+	struct gpio_litex_data *data = dev->data;
+
+	return gpio_manage_callback(&data->cb, callback, set);
+}
+
 static int gpio_litex_pin_interrupt_configure(const struct device *dev,
 					      gpio_pin_t pin,
 					      enum gpio_int_mode mode,
 					      enum gpio_int_trig trig)
 {
-	int ret = 0;
-
-	if (mode != GPIO_INT_MODE_DISABLED) {
-		ret = -ENOTSUP;
+	const struct gpio_litex_cfg *gpio_config = DEV_GPIO_CFG(dev);
+	int ret = -1;
+
+	if (mode == GPIO_INT_MODE_EDGE) {
+		uint8_t ev_enabled = litex_read(GPIO_EV_ENABLE(dev),
+				gpio_config->reg_size);
+		litex_write(GPIO_EV_ENABLE(dev), gpio_config->reg_size,
+			    ev_enabled | BIT(pin));
+		ret = 0;
 	}
```

Comment:
Sorry, I missed it before. It looks like the driver supports edge interrupts on both edges only. We should return `-ENOTSUP` in case user tries to configure level interrupt or edge interrupt on a single edge.

It would be good to test the driver with `tests/drivers/gpio/gpio_basic_api/` test.

Reply:
I changed `int ret = -1` to `int ret= -ENOTSUP` so there will be `-ENOTSUP` returned always if user tries to configure interrupt different than edge interrupt. Let me know if that's okay.

I don't think I am able to test this with `gpio_basic_api` since it requires to have `in-gpios` and `out-gpios` on the same GPIO port. I am using Arty A7 development board with LiteX SoC Builder. Unfortunately LiteX does not support interrupts on `GPIOInOut`. It is supported on `GPIOIn` only. That's what I was testing - multiple interrupts on multiple `GPIOIn` modules in `GPIO_INT_EDGE_RISING` or `GPIO_INT_EDGE_FALLING` modes. 
I've also tested it with `GPIO_INT_EDGE_BOTH` mode but since it is not supported by LiteX it works just like falling/rising edge mode depending on IRQ setup in LiteX `GPIOIn` module.

Reply:
According to my previous coment I decided to add a condition `if(gpio_config->port_is_output == true)` which returns `-ENOTSUP` if port is an output port. 
I did get rid of `int ret` as well so I think it is more clear what function returns now.